app/src/ActionModal/components/DelegationModal.vue

```diff
@@ -355,6 +355,10 @@ export default {
 
       // update registered topics for emails as the validator set changed
       this.$store.dispatch("updateEmailRegistrations")
+      // update the role of the user as it might change after bonding the first time
+      if (this.currentNetwork.network_type === "polkadot") {
```

Comment:
anytime i see network specific stuff it feels wrong

Reply:
There are a bunch of network type logics. This endpoint actually works for Cosmos as well. Bit we safe a request. Happy to discuss further. But will leave this in for now.